src/microbit/test/test_microbit_model.py

```diff
@@ -0,0 +1,24 @@
+import time
+
+import pytest
+from unittest import mock
+from ..model.microbit_model import MicrobitModel
+
+
+class TestMicrobitModel(object):
+    def setup_method(self):
+        self.mb = MicrobitModel()
+
+    @pytest.mark.parametrize("value", [9, 30, 1999])
+    def test_sleep(self, value):
+        time.sleep = mock.Mock()
+        self.mb.sleep(value)
+        time.sleep.assert_called_with(value / 1000)
+
+    def test_running_time(self):
+        mock_start_time = 10
+        mock_end_time = 300
+        self.mb._MicrobitModel__start_time = mock_start_time

```

Comment:
Just FYI, (you might already be familiar with this) you might have the constructor be something like:
```
class MicrobitModel:
    def __init__(self, start_time=time.time()):
        # State in the Python process
        self.button_a = Button()
        self.button_b = Button()
        self.__start_time = start_time
```
By exposing the "hidden" parameter, it is easier to test that functionality of the class.
Because we want to be 1:1 with the micro:bit library, we can't do that this time though.

Reply:
ooo, i did not know that. Thank you :D
I won't be changing it but it is nice to know :)